conan.cmake

```diff
@@ -357,7 +357,8 @@ function(conan_cmake_install)
     endif()
     set(CONAN_OPTIONS "")
     if(ARGUMENTS_CONANFILE)
-      set(CONANFILE ${CMAKE_CURRENT_SOURCE_DIR}/${ARGUMENTS_CONANFILE})
+      get_filename_component(CONANFILE ${ARGUMENTS_CONANFILE}
+                             ABSOLUTE BASE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
```

Comment:
Note that the default for BASE_DIR is ${CMAKE_CURRENT_SOURCE_DIR}, but explicit is better than implicit.